connexion/lifecycle.py

```diff
@@ -28,6 +28,10 @@ def content_type(self) -> str:
     def mimetype(self) -> str:
         raise NotImplementedError
 
+    @property
+    def headers(self) -> t.Any:
```

Comment:
Only noticed the `__getattr__` function now, seems like the `spec_set` argument of `MagicMock` isn't able to properly detect this, leading to an error in the tests